spec/acceptance/api/internal/services_api_spec.rb

```diff
@@ -67,12 +67,10 @@
           expect(response_json['service']['state']).to eq state.to_s
         end
       end
-      context 'with an service that has invalid state in db' do
+      context 'when state is set to an invalid state' do
+        let(:state) { :not_valid_state }
         let(:id) { otherid }
-        let(:storage) { ThreeScale::Backend::Storage.instance }
-        example 'Get Service by ID should return inactive state' do
-          storage.set ThreeScale::Backend::Service.storage_key(id, 'state'), 'invalid_state'
```

Comment:
if you remove direct db update, db registry will not have invalid state

Reply:
That's not what it is being tested anymore. Now it tests that when setgting an invalid state it returns suspended:
```
      context 'when state is set to an invalid state' do
```

The invalid state in the DB test has been moved to spec/service.rb

Reply:
This is end to end acceptance test. ```service_spec.rb``` test scope is more functional, unittest. The purpose of the test was: 'What do I get when I request via API a service and in database state value is corrupted". The unittest in ```service_spec.rb``` does not answer that.

In the way it is now, we know from other test in ```service_spec``` that saving service with invalid state attribute, does not propagate to database and state is set as ```:suspended```, the same situation as the previous tests ```when state is suspended```. There is no new use case being tested as it is now.

Reply:
I think that having knowledge on what happens to the database from the point of view of the API specs should not be tested, although I understand the need of looking at what happens when you try to obtain a service that has no db state record. I tried to  cover the specific case of what happens in the DB in the Service spec, when the Service.load_by_id is called (that it's what is called by the API get method at the end)

Reply:
There is no way to save service info to database leaving state empty using current version of service class. For the end to end acceptance test, we need to do it manually. I agree with you for functional unittests.

Reply:
We want to specify the internal API here, not do e2e tests. The non-existent state in storage issue is an implementation + storage migration detail that is covered with the unit test, albeit not an integration test. But an integration test covering this is of no value if the unit tests already prove that the storage value does never leak outside, so it becomes an internal detail that we should not care about at that level.